clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -449,6 +449,17 @@ static void InitializeStandardPredefinedMacros(const TargetInfo &TI,
         Builder.defineMacro("__FAST_RELAXED_MATH__");
     }
   }
+
+  // TODO: How do I enable this only if -sycl has been set?
+  switch (LangOpts.getSYCLVersion()) {
+    case LangOptions::SYCLVersionList::sycl_121:
+      Builder.defineMacro("CL_SYCL_LANGUAGE_VERSION", "121");
+      break;
+    default:
+      llvm_unreachable("Unsupported SYCL version");
```

Comment:
-sycl is a driver option IIRC, the frontend option is for device mode only.

But here you are in cc1 mode, no one uses it directly unless you know what you are doing. My suggestion is rely on the driver to provide the value to the frontend if sycl is enabled.

Then here, default should be understood as "no SYCL mode enable" and therefore do nothing